### PR TITLE
Correct zero length check expressions

### PR DESCRIPTION
**Pull request by MikeAlhayek**
I noticed many invalid expression used in OC when checking for a zero elements in a collection.

Here is explanation

```
List<string> items = null;

if(items?.Count == 0)
{
	// This will evaluate false since null does not equals 0.
}

if(items == null || items.Count == 0)
{
	// This will evaluate true since items either null OR is not null and it's count is 0.
}

if(items?.Count > 0)
{
	// This check is okay because null items will return false and if the Count is 0 will also false.
}
```

@Piedone it would be nice to add an analyzer to guard against such a count check. Maybe related to #7950

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Flows/Controllers/AdminController.cs [hunk 1]
```diff
@@ -111,7 +111,7 @@ private async Task<IEnumerable<ContentTypeDefinition>> GetContainedContentTypesA
         {
             var settings = (await _contentDefinitionManager.GetTypeDefinitionAsync(contentType))?.Parts.SingleOrDefault(x => x.Name == partName)?.GetSettings<FlowPartSettings>();
 
-            if (settings?.ContainedContentTypes?.Length == 0)
+            if (settings?.ContainedContentTypes == null || settings.ContainedContentTypes.Length == 0)
```
Comment (Piedone) [inline src/OrchardCore.Modules/OrchardCore.Flows/Controllers/AdminController.cs:114]: FYI I checked if the more succinct `items?.Any() == false` pattern would work. It wouldn't, as it (I don't understand why) produces `null`; and it also violates [CA1860](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1860), nudging you towards `items?.Count == 0`. So, that's not an option.

**Comment by MikeAlhayek** [comment]
@gvkries feel free to leave code review and approve PRs when you visit them

File: src/OrchardCore/OrchardCore.Search.AzureAI.Core/Recipes/AzureAISearchIndexSettingsStep.cs [hunk 2]
```diff
@@ -21,18 +22,21 @@ public sealed class AzureAISearchIndexSettingsStep : IRecipeStepHandler
...
         AzureAISearchIndexManager indexManager,
         AzureAIIndexDocumentManager azureAIIndexDocumentManager,
         AzureAISearchIndexSettingsService azureAISearchIndexSettingsService,
+        ILogger<AzureAISearchIndexSettingsStep> logger,
         IStringLocalizer<AzureAISearchIndexSettingsStep> stringLocalizer)
     {
         _indexManager = indexManager;
         _azureAIIndexDocumentManager = azureAIIndexDocumentManager;
         _azureAISearchIndexSettingsService = azureAISearchIndexSettingsService;
+        _logger = logger;
```
Comment (Piedone) [inline src/OrchardCore/OrchardCore.Search.AzureAI.Core/Recipes/AzureAISearchIndexSettingsStep.cs:39]: Did you intend to use this to log something?

**Review by Piedone — CHANGES_REQUESTED** [review]

File: src/OrchardCore/OrchardCore.Search.AzureAI.Core/Recipes/AzureAISearchIndexSettingsStep.cs (hunk 2 above)
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Search.AzureAI.Core/Recipes/AzureAISearchIndexSettingsStep.cs]: done.

**Pull request merged**